### PR TITLE
feat: Split by Title (#441)

### PR DESCRIPTION
**Pull request by cxumol**
### Overview

Splitting output video file according to stream room title proposed as a feature at #441. 

A new configure option is added accordingly. In the design level of BililiveRecorder, its splitting configuration strategy was selecting one option from many choices. It worked well when users don't usually need splitting by both file size and video length at the same time. Now if we want to introduce splitting by video stream title as a new cutting mode, I aware it's reasonable to enable this mode work together with every existing cutting mode.

Proposing in this PR, existing splitting modes remain in the same behaviours to the user, and "Split By Title" can perform as an additional layer over existing splitting modes.

### Core 
- Add `CuttingMode.ByTitle` to `BililiveRecorder.Core.Config`
- Update `BililiveRecorder.Core.Config.CuttingMode` from `enum` to `[System.Flags] enum` to fulfill increased complexity level of `CuttingMode` while maintaining compatibility to earlier version
- Add logic to split output file if room title changed to `BililiveRecorder.Core.Room` as per `Room_PropertyChanged` event

### WPF
- Add `FlagsEnumToBooleanConverter` to `BililiveRecorder.WPF.Converters`, with relevant implementations and references.
- Add WPF XAML GUI elements, where `CuttingMode.ByTitle` displayed as `<ui:ToggleSwitch>`
- Multi-language translations for new GUI elements

### Test
- Changed unit test file according to the change on BililiveRecorder.Core.Config

Please let me know if there are any concerns, warnings or errors.

### CODE REVIEW

File: BililiveRecorder.WPF/Properties/Strings.resx [hunk 1]
```diff
@@ -490,7 +490,10 @@
     <value>根据视频时间自动分段</value>
   </data>
   <data name="Settings_Splitting_RadioButton_Disabled" xml:space="preserve">
-    <value>不自动分段</value>
+    <value>不要根据文件大小/视频时间自动分段</value>
+  </data>
+  <data name="Settings_Splitting_ToggleSwitch_ByTitle" xml:space="preserve">
+    <value>根据直播间标题自动分段</value>
```
Comment (Genteure) [inline BililiveRecorder.WPF/Properties/Strings.resx:496]: 建议改成“直播间标题修改后自动分段”

**Review by Genteure — COMMENTED** [review]
感谢你的 PR！

我感觉与其改 `CuttingMode` 的值的定义，新增加一个设置项更合理也更简单一点。

假设前提是*尽量*保持 2.x 版本 WebUI 和录播姬本体 API 的兼容。
修改值的定义后，如果启用了标题变动后分段，旧版本 WebUI 就处理不了现有的自动分段设置了，因为值在预期范围外。
新增新的设置项，旧版本 WebUI 只是不显示新的设置，不影响修改和显示现有的自动分段设置。

新增设置项的方法是，修改 `config_gen/data.ts`，然后在 `config_gen` 目录里运行 `npm build c`。

git submodule 不用在 PR 里更新，我合并后发新版本之前再更新一下就行。

**Comment by cxumol** [comment]
多谢指点! 兼容性方面的考虑十分周到, 我会按照建议修改｡ git submodule 也会注意排除｡ 有意开新分支修改再发 PR, 所以这个 PR 就先关了

**Comment by cxumol** [comment]
> git submodule 不用在 PR 里更新

不知道 git submodule 要怎么设置才能避免 diff 信息被 commit 进上层仓库｡ stackoverflow 上有几个说要改 .gitmodules 的设置, 给子模块加个ignore = all, 我又怕加 ignore 会有别的副作用｡

**Comment by Genteure** [comment]
在父仓库 commit 之前不 stage submodule，就是不 `git add` submodule 的改动。实在不行你 push 上来，我合并之前给 revert 掉就行了。

**Comment by cxumol** [comment]
我之前把 submodule 给 git rm 掉然后 push, 结果 CI 上来发现 No such file or directory, 所以不得不重新加上 submodule 后 push 才能过 CI. 总之这把我 submodule 的改动已经迫不得已提交了, 如果能通过 PR, 还得请你麻烦多操作一步

**Pull request closed without merging**